### DISCUSSION TITLE
Context usage in public facing APIs of `XlsArea`

### DISCUSSION THREAD

**Discussion by tobHai** [jxlsteam/jxls#340] [Q&A]
I'm currently updating JXLS from version 2 to version 3 in one of my projects. 

The migration guide states that _Map<String, Object> replaces Context_. For most of my usages this change was easy to implement. 

But one of my use cases also uses `XlsArea` and calls `public Size applyAt(CellRef cellRef, Context context` on it. 
I wonder what's the best approach to create a context object for the usage with this API?

Is it fine to use `new ContextImpl()` for now?

Thanks!

**Comment by gastendonk** [comment]
I can't answer it 100% because you didn't give enough information.

There is no guarantee that ContextImpl will still be available in future versions. Therefore, use it at your own risk. ;-) Therefore, the best solution is to implement Context yourself.

**Comment by tobHai** [comment]
Thanks for the quick response.  

I was a bit confused by the comment in `Context.java` stating `Jxls context interface (Jxls internal class)`. 
If the `Context` interface is internal, shouldn't the public methods in `XlsArea` for accept implementations of `PublicContext`?
